### PR TITLE
Add find-visible-column-for-legacy-ref conversion helper

### PR DESCRIPTION
**Pull request by metamben**
Part of #33022.

### CODE REVIEW

**Review by bshepherdson — CHANGES_REQUESTED** [review]
This feels over-specialized to me. I think this should be a TS helper that calls `Lib.visible_columns` and `Lib.find_column_indexes_from_legacy_refs`.

I see on reflection that `Lib.visible_columns` doesn't exist in master, only the [escalation branch](https://github.com/metabase/metabase/blob/mblib-column-settings-round-trip/src/metabase/lib/js.cljs#L437).

That would be easy to split out into a separate PR along with the TS helper I propose - should I do that now?

**Comment by metamben** [comment]
> That would be easy to split out into a separate PR along with the TS helper I propose - should I do that now?

Yes, please.

**Review by bshepherdson — APPROVED** [review]
Extracting enough of #32698 to support this is taking longer than I'd hoped.

If this is blocking other work, go ahead and submit it. I'll keep working on breaking up that mega-PR in any case; it's way too big and splitting out some pieces is a good idea anyway.

File: frontend/src/metabase-lib/fields.ts [hunk 1]
```diff
@@ -30,3 +31,11 @@ export function fieldableColumns(
 export function _fieldId(column: ColumnMetadata): number | null {
   return ML.field_id(column);
 }
+
+export function findVisibleColumnForLegacyRef(
+  query: Query,
+  stageIndex: number,
+  fieldRef: FieldReference,
+): ColumnMetadata | null {
```
Comment (iethree) [inline frontend/src/metabase-lib/fields.ts:39]: I think we need a `ColumnWithOperators` from this, or maybe even more helpfully a way to turn a `ColumnMetadata` column into a `ColumnWithOperators` ?

**Pull request merged**